sigar_linux.go

```diff
@@ -216,6 +217,146 @@ func (self *NetIfaceList) Get() error {
 	return err
 }
 
+func (self *NetTcpConnList) Get() error {
+	list, err := readConnList(Procd+"/net/tcp", 4, 17)
+	if err != nil {
+		return err
+	}
+	self.List = list
+	return nil
+}
+
+func (self *NetUdpConnList) Get() error {
+	list, err := readConnList(Procd+"/net/udp", 4, 13)
+	if err != nil {
+		return err
+	}
+	self.List = list
+	return nil
+}
+
+func (self *NetRawConnList) Get() error {
+	list, err := readConnList(Procd+"/net/raw", 4, 13)
+	if err != nil {
+		return err
+	}
+	self.List = list
+	return nil
+}
+
+func (self *NetTcpV6ConnList) Get() error {
+	list, err := readConnList(Procd+"/net/tcp6", 16, 17)
+	if err != nil {
+		return err
+	}
+	self.List = list
+	return nil
+}
+
+func (self *NetUdpV6ConnList) Get() error {
+	list, err := readConnList(Procd+"/net/udp6", 16, 13)
+	if err != nil {
+		return err
+	}
+	self.List = list
+	return nil
+}
+
+func (self *NetRawV6ConnList) Get() error {
+	list, err := readConnList(Procd+"/net/raw6", 16, 13)
+	if err != nil {
+		return err
+	}
+	self.List = list
+	return nil
+}
+
+/* Reads the format of the /proc/net/<proto> files, which have 2 header lines and a
+   list of open connections. Different protocols have different numbers of trailing fields,
+   but the first 5 are the same. */
+func readConnList(listFile string, ipSizeBytes, numFields int) ([]NetConn, error) {
+	connList := make([]NetConn, 0)
+	err := readFile(listFile, func(line string) bool {
+		fields := strings.Fields(line)
+		if len(fields) != numFields {
+			return true
+		}
+		// Skip the header, only take lines where the first field is <number>:
+		if fields[0][len(fields[0])-1] != ':' {
+			return true
+		}
+
+		var err error
+		var conn NetConn
+		conn.LocalAddr, conn.LocalPort, err = readConnIp(fields[1], ipSizeBytes)
+		if err != nil {
+			return true
+		}
+
+		conn.RemoteAddr, conn.RemotePort, err = readConnIp(fields[2], ipSizeBytes)
+		if err != nil {
+			return true
+		}
+
+		status, err := strconv.ParseInt(fields[3], 16, 8)
+		if err != nil {
+			return true
+		}
+
+		conn.Status = NetConnState(status)
+		queues := strings.Split(fields[4], ":")
+		if len(queues) != 2 {
+			return true
+		}
+
+		conn.SendQueue, err = strtoull(queues[0])
+		if err != nil {
+			return true
+		}
+
+		conn.RecvQueue, err = strtoull(queues[1])
+		if err != nil {
+			return true
+		}
+
+		connList = append(connList, conn)
+		return true
+	})
+	return connList, err
+}
+
+/* Decode an IP:port pair, with either a 16-byte or 4-byte IP as specified by lenBytes
+   encoded in network order as hexidecimal characters ex. 0100007F -> 127.0.0.1 */
```

Comment:
This comment seems off as network order would be big endian, e.g. `7F000001` for `127.0.0.1`.